File: src/main/java/com/rimlang/rim/syntax/VariableDefinitionSyntaxNode.java
package com.rimlang.rim.syntax;

import com.rimlang.rim.lexer.StringToken;

public class VariableDefinitionSyntaxNode extends SyntaxNode {
    private final StringToken identifier;
    private final GenericExpression initialValue;

    public VariableDefinitionSyntaxNode(StringToken identifier, GenericExpression initialValue) {
        this.identifier = identifier;
        this.initialValue = initialValue;
    }

    public StringToken getIdentifier() {
        return identifier;
    }

    public GenericExpression getInitialValue() {
        return initialValue;
    }
}

File: src/main/kotlin/com/rimlang/rim/syntax/node/syntax/VariableDefinitionSyntaxNode.kt
package com.rimlang.rim.syntax.node.syntax

import com.rimlang.rim.lexer.token.StringToken
import com.rimlang.rim.lexer.token.Token
import com.rimlang.rim.syntax.node.expression.GenericExpression

data class VariableDefinitionSyntaxNode(
    val identifier: StringToken,
    val initialValue: GenericExpression,
    override val firstToken: Token
) : SyntaxNode(firstToken)